packages/core/contracts/financial-templates/expiring-multiparty/Liquidatable.sol

```diff
@@ -504,6 +504,28 @@ contract Liquidatable is PricelessPositionManager {
         return liquidations[sponsor];
     }
 
+    /**
+     * @notice Accessor method to calculate a transformed collateral requirement using the finanical product library
+      specified during contract deployment. If no library was provided then no modification to the collateral requirement is done.
+     * @param price input price used as an input to transform the collateral requirement.
+     * @return transformedCollateralRequirement collateral requirement with transformation applied to it.
+     * @dev This method should never revert.
+     */
+    function transformCollateralRequirement(FixedPoint.Unsigned memory price)
+        public
+        view
+        returns (FixedPoint.Unsigned memory)
+    {
+        if (address(financialProductLibrary) == address(0)) return collateralRequirement;
```

Comment:
nit: if we're already using a try-catch here, would it make sense to _not_ do this explicit check and just let the try-catch handle the fallback?

Reply:
I also think the try-catch is subtly really cool here because it allows people to pass in financial product libraries that only implement the methods that they actually want to override.

Reply:
this was intentional :) I initially did it the way you propose, (and the same is true with the price transformation) but I get an error as follows:
```
     Error: Transaction reverted: function call to a non-contract account
```

This occurs if the `financialProductLibrary` is set to an EOA, such as the address(0). This check would ensure that it would not revert at this point.

However, you did make me realize that I did not have a test for setting the `financialProductLibrary` to another EOA address! I tested this and indeed the same revert happens. 

I've updated the function to first check if the library is set to a contract. If it is not, then it does nothing. If it is, then it enters into the try-catch logic block.

I've also updated the previous library for price transformations to use the same logic.

PTAL.